### PR TITLE
avoid using for loop in space psds; fix typo

### PR DESCRIPTION
**Pull request by HumphreyWang**
@WuShichao I removed all for loops from the space PSD functions to avoid unnecessary time costs. There are still duplicate lines though, not sure if you want to optimize them further.

### CODE REVIEW

**Comment by ahnitz** [comment]
@HumphreyWang Thank you for fixing this!

File: pycbc/psd/analytical_space.py [hunk 1]
```diff
@@ -29,7 +29,7 @@
 """
 
 import numpy as np
-from astropy import constants
+from scipy.constants import c
```
Comment (ahnitz) [inline pycbc/psd/analytical_space.py:32]: In general we should preference astropy for constants. Is there a reason to prefer the scipy version?

Comment (HumphreyWang) [reply pycbc/psd/analytical_space.py:32]: When using `astropy.constants.c`, there will be a non-critical warning said "Cannot find reference 'c' in '__init__.py'", and also it needs to be converted to float type by additionally saying `c.value`. It's OK to change back, I'll do it later!

**Comment by WuShichao** [comment]
@HumphreyWang Thanks for the optimization of the PSD module; I will review that soon. I will also add other space-borne detectors' PSD in a new PR soon. I remember there is a reason why I didn't forbid for loop here; I will test if all is functional.

**Review by WuShichao — APPROVED** [review]
I have tested all functions in this module:
image
image
Looks good, but I notice there are small but noticeable changes in the T channel curve in some tests, maybe due to different frequency bins.

**Pull request merged**